### PR TITLE
flight-roster-endpoint

### PR DESCRIPTION
**Pull request by nehaabrol87**
@jonathanbarton @morrissinger  :  This is my first stab at the flight roster endpoint which returns a list of flights.
I will be writing tests now but wanted to put this out to be reviewed so that I can incorporate if there are any review comments .

**Couple of things**
1.I think callSign and flightNumber should have a compound index which should be unique .
Also I think call signs should be unique to avoid confusion.Let me know what you guys think and I will add them.
2.I created a gomake db locally to test the endpoints .Maybe I can push out a json file and we can start sharing it for testing purposes.Let me know your thoughts about that.

### CODE REVIEW

**Comment by nehaabrol87** [comment]
Also I think the endpoint should be /flights instead of flight , since it was /flight in the initial app I did not want to change it without discussion/confirmation.

**Comment by morrissinger** [comment]
#10 includes a container that runs mongo, so by starting with docker-compose, both the API and Mongo will start together. The name of the DB comes from an environmental variable, which is specified in docker-compose.yaml, so we will be doing away with the different configuration files under config/env.

The rest of the commit LGTM.

**Comment by nehaabrol87** [comment]
@morrissinger  Yes yes.Ooops I am docker noob . So I understand that I need to build the gomake/api (docker build -t gomake/api .) to containerize the api which will be run when I run  docker compose up command but what about gomake/mongo don't  we need to containerize that as well as we have that image in the yml file as well.

**Comment by morrissinger** [comment]
Sorry for the friction on this. The mongo container is coming from https://github.com/jonathanbarton/gomake-api/blob/master/mongo/Dockerfile. You will have to build gomake/mongo from that Dockerfile. I am going to try to push that image to our private Docker registry to eliminate that step.

File: server/tests/flight.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import request from 'supertest-as-promised';
```
Comment (nehaabrol87) [inline server/tests/flight.test.js]: @jonathanbarton  @morrissinger  I have added the tests for this endpoint.
I was planning to use https://github.com/williamkapke/mongo-mock for mocking mongoDb for the tests.Let me know if you guys have another mock db in mind.

Comment (jonathanbarton) [inline server/tests/flight.test.js]: sounds alright to me. I was also looking at https://www.npmjs.com/package/drakov to mock endpoints using the api-blueprint document, but wont need it for the time being i suppose.

Comment (jonathanbarton) [inline server/tests/flight.test.js]: we should make sure we keep static data files for the db data mocking in a well organized folder structure.

Comment (nehaabrol87) [inline server/tests/flight.test.js]: yes yes maybe we can jam on that together so that it is consistent.

**Pull request merged**